webapp/login/views.py

```diff
@@ -26,7 +26,11 @@ def login_handler():
     if authentication.is_authenticated(flask.session):
         return flask.redirect(open_id.get_next_url())
 
-    root = authentication.request_macaroon()
+    try:
+        root = authentication.request_macaroon()
+    except Exception:
```

Comment:
This is extremely broad. It will, for example, log you out if there's a brief network glitch. Perhaps limit the redirect to when you get a 401 from SSO?

Reply:
Now redirecting to an error page on anything else than `401` otherwise logout